File: EL11.java
interface Figure {
    String getColor();
    double perimetro();
    double area();
}

class Triangle implements Figure {
    private String color;
    private double base, altura, lado1, lado2, lado3;

    public Triangle(String color, double base, double altura, double lado1, double lado2, double lado3) {
        this.color = color;
        this.base = base;
        this.altura = altura;
        this.lado1 = lado1;
        this.lado2 = lado2;
        this.lado3 = lado3;
    }

    @Override
    public String getColor() {
        return color;
    }

    @Override
    public double perimetro() {
        return lado1 + lado2 + lado3;
    }

    @Override
    public double area() {
        return (base * altura) / 2;
    }
}

class Circle implements Figure {
    private String color;
    private double radio;

    public Circle(String color, double radio) {
        this.color = color;
        this.radio = radio;
    }

    @Override
    public String getColor() {
        return color;
    }

    @Override
    public double perimetro() {
        return 2 * Math.PI * radio;
    }

    @Override
    public double area() {
        return Math.PI * Math.pow(radio, 2);
    }
}

class Rectangle implements Figure {
    private String color;
    private double largo, ancho;

    public Rectangle(String color, double largo, double ancho) {
        this.color = color;
        this.largo = largo;
        this.ancho = ancho;
    }

    @Override
    public String getColor() {
        return color;
    }

    @Override
    public double perimetro() {
        return 2 * (largo + ancho);
    }

    @Override
    public double area() {
        return largo * ancho;
    }
}

class Hexagon implements Figure {
    private String color;
    private double lado;

    public Hexagon(String color, double lado) {
        this.color = color;
        this.lado = lado;
    }

    @Override
    public String getColor() {
        return color;
    }

    @Override
    public double perimetro() {
        return 6 * lado;
    }

    @Override
    public double area() {
        return (3 * Math.sqrt(3) * Math.pow(lado, 2)) / 2;
    }
}

public class Main {
    public static void main(String[] args) {
        Figure[] shapes = {
            new Triangle("red", 3, 4, 3, 4, 5),
            new Circle("blue", 5),
            new Rectangle("green", 4, 6),
            new Hexagon("yellow", 2)
        };

        for (Figure shape : shapes) {
            System.out.println("Color: " + shape.getColor() + ", Perimeter: " + shape.perimetro() + ", Area: " + shape.area());
        }
    }
}

File: EL11.py
from abc import ABC, abstractmethod
import math

class Figure(ABC):
    @abstractmethod
    def getColor(self):
        pass

    @abstractmethod
    def perimetro(self):
        pass
    
    @abstractmethod
    def area(self):
        pass

class Triangle(Figure):
    def __init__(self, color, base, altura, lado1, lado2, lado3):
        self.color = color
        self.base = base
        self.altura = altura
        self.lado1 = lado1
        self.lado2 = lado2
        self.lado3 = lado3
    
    def getColor(self):
        return self.color
    
    def perimetro(self):
        return self.lado1 + self.lado2 + self.lado3
    
    def area(self):
        return (self.base * self.altura) / 2

class Circle(Figure):
    def __init__(self, color, radio):
        self.color = color
        self.radio = radio
    
    def getColor(self):
        return self.color
    
    def perimetro(self):
        return 2 * math.pi * self.radio
    
    def area(self):
        return math.pi * self.radio ** 2

class Rectangle(Figure):
    def __init__(self, color, largo, ancho):
        self.color = color
        self.largo = largo
        self.ancho = ancho
    
    def getColor(self):
        return self.color
    
    def perimetro(self):
        return 2 * (self.largo + self.ancho)
    
    def area(self):
        return self.largo * self.ancho

class Hexagon(Figure):
    def __init__(self, color, lado):
        self.color = color
        self.lado = lado
    
    def getColor(self):
        return self.color
    
    def perimetro(self):
        return 6 * self.lado
    
    def area(self):
        return (3 * math.sqrt(3) * (self.lado ** 2)) / 2

# Demostración de polimorfismo
shapes = [
    Triangle("red", 3, 4, 3, 4, 5),
    Circle("blue", 5),
    Rectangle("green", 4, 6),
    Hexagon("yellow", 2)
]

for shape in shapes:
    print(f"Color: {shape.getColor()}, Perimeter: {shape.perimetro()}, Area: {shape.area()}")
